test/xds/xds_client_federation_test.go

```diff
@@ -140,3 +143,133 @@ func (s) TestClientSideFederation(t *testing.T) {
 		t.Fatalf("rpc EmptyCall() failed: %v", err)
 	}
 }
+
+// TestFederation_UnknownAuthorityInDialTarget tests the case where a ClientConn
+// is created with a dial target containing an authority which is not specified
+// in the bootstrap configuration. The test verifies that RPCs on the ClientConn
+// fail with an appropriate error.
+func (s) TestFederation_UnknownAuthorityInDialTarget(t *testing.T) {
+	oldXDSFederation := envconfig.XDSFederation
+	envconfig.XDSFederation = true
+	defer func() { envconfig.XDSFederation = oldXDSFederation }()
+
+	// Setting up the management server it not *really* required for this test
+	// case. All we need is a bootstrap configuration which does not contain the
+	// authority mentioned in the dial target. But setting up the management
+	// server and actually making an RPC ensures that the xDS client is
+	// configured properly, and when we dial with an unknown authority in the
+	// next step, we can be sure that the error we receive is legitimate.
+	managementServer, nodeID, _, resolver, cleanup1 := e2e.SetupManagementServer(t, e2e.ManagementServerOptions{})
+	defer cleanup1()
+
+	port, cleanup2 := startTestService(t, nil)
+	defer cleanup2()
+
+	const serviceName = "my-service-client-side-xds"
+	resources := e2e.DefaultClientResources(e2e.ResourceParams{
+		DialTarget: serviceName,
+		NodeID:     nodeID,
+		Host:       "localhost",
+		Port:       port,
+		SecLevel:   e2e.SecurityLevelNone,
+	})
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	if err := managementServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create a ClientConn and make a successful RPC.
+	target := fmt.Sprintf("xds:///%s", serviceName)
+	cc, err := grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(resolver))
+	if err != nil {
+		t.Fatalf("Dialing target %q: %v", target, err)
+	}
+	defer cc.Close()
+	t.Log("Created ClientConn to test service")
+
+	client := testpb.NewTestServiceClient(cc)
+	if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
+		t.Fatalf("EmptyCall() RPC: %v", err)
+	}
+	t.Log("Successfully performed an EmptyCall RPC")
+
+	target = fmt.Sprintf("xds://unknown-authority/%s", serviceName)
+	t.Logf("Dialing target %q with unknown authority which is expected to fail", target)
+	_, err = grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(resolver))
+	if err == nil || !strings.Contains(err.Error(), `authority "unknown-authority" is not found in the bootstrap file`) {
+		t.Fatal("grpc.Dial() for target with unknown authority succeeded when expected to fail")
```

Comment:
Not necessarily; it could be it errored with some other error.  Please log the actual error.

Reply:
Done.